libwebauthn/src/transport/nfc/channel.rs

```diff
@@ -0,0 +1,335 @@
+            fido2 = true;
+            // NOTE: Yubikeys seem to ignore this part of the specification and always return U2F_V2, even if U2F-NFC is disabled.
+        } else if response == b"U2F_V2" {
+            //     If the authenticator implements CTAP1/U2F, the version information SHALL be the string "U2F_V2", or 0x5532465f5632, to maintain backwards-compatibility with CTAP1/U2F-only clients.
+            u2f = true;
+            //     If the authenticator implements both CTAP1/U2F and CTAP2, the version information SHALL be the string "U2F_V2", or 0x5532465f5632, to maintain backwards-compatibility with CTAP1/U2F-only clients. CTAP2-aware clients MAY then issue a CTAP authenticatorGetInfo command to determine if the device supports CTAP2 or not.
+            fido2 = self.ctap2_get_info().await.is_ok();
+        }
+
+        self.supported = SupportedProtocols { u2f, fido2 };
+
+        Ok(())
+    }
+
+    fn handle_in_ctx(
+        &mut self,
+        ctx: Ctx,
+        command_buf: &[u8],
+        buf: &mut [u8],
+    ) -> Result<usize, NfcError> {
+        let res = self.delegate.handle_in_ctx(ctx, command_buf, buf)?;
+        Ok(res)
+    }
+
+    pub fn handle<'a>(
+        &'a mut self,
+        ctx: Ctx,
+        command: impl Into<Command<'a>>,
+    ) -> Result<Vec<u8>, NfcError> {
+        let command = command.into();
+        let command_buf = Vec::from(command);
+
+        let mut buf = [0u8; 1024];
+        let mut rapdu = Vec::new();
+
+        let len = self.handle_in_ctx(ctx, &command_buf, &mut buf)?;
+        let mut resp = Response::from(&buf[..len]);
+
+        let (mut sw1, mut sw2) = resp.trailer;
+        rapdu.extend_from_slice(resp.payload);
+
+        while sw1 == SW1_MORE_DATA {
+            let get_response_cmd = command_get_response(0x00, 0x00, sw2);
+            let get_response_buf = Vec::from(get_response_cmd);
+            let len = self.handle_in_ctx(ctx, &get_response_buf, &mut buf)?;
+            resp = Response::from(&buf[..len]);
+            (sw1, sw2) = resp.trailer;
+            rapdu.extend_from_slice(resp.payload);
+        }
+
+        rapdu.extend_from_slice(&[sw1, sw2]);
+        Result::from(Response::from(rapdu.as_slice()))
+            .map(|p| p.to_vec())
+            .map_err(|e| {
+                trace!("map_err {:?}", e);
+                apdu::Error::from(e).into()
+            })
+    }
+
+    #[instrument(skip_all)]
+    pub async fn blink_and_wait_for_user_presence(
+        &mut self,
+        _timeout: Duration,
+    ) -> Result<bool, Error> {
+        unimplemented!()
+    }
+}
+
+#[async_trait]
+impl<Ctx> Channel for NfcChannel<Ctx>
+where
+    Ctx: Copy + Send + Sync + fmt::Debug + Display,
+{
+    type UxUpdate = UvUpdate;
+
+    async fn supported_protocols(&self) -> Result<SupportedProtocols, Error> {
+        Ok(self.supported)
+    }
+
+    async fn status(&self) -> ChannelStatus {
+        self.status
+    }
+
+    async fn close(&mut self) {
+        todo!("close")
```

Comment:
Leftover here, I think. Although we shouldn't panic, I'm not sure if we're actually using `close` anywhere (best addressed separately).